src/main/java/seedu/address/logic/commands/EditApplicantCommand.java

```diff
@@ -0,0 +1,151 @@
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_LISTING_DISPLAYED_INDEX);
+        }
+
+        Listing listingToEditApplicantFrom = lastShownList.get(targetIndex.getZeroBased());
+
+        Optional<Applicant> oldApplicant = getOldApplicantObject(listingToEditApplicantFrom);
+
+        if (listingToEditApplicantFrom.getApplicants().stream().filter(
+                        applicant -> applicant.equals(newApplicant) && applicant.hashCode() == newApplicant.hashCode())
+                .count() > 0) {
+            throw new CommandException(MESSAGE_DUPLICATE_APPLICANT);
+        }
+
+        if (oldApplicant.isEmpty()) {
+            throw new CommandException(String.format(MESSAGE_APPLICANT_NOT_FOUND, oldApplicantId,
+                    listingToEditApplicantFrom.getTitle()));
+        }
+
+        Listing editedListing = createEditedListing(listingToEditApplicantFrom,
+                oldApplicant.get(), newApplicant);
+
+        model.setListing(listingToEditApplicantFrom, editedListing);
+
+        return new CommandResult(String.format(MESSAGE_SUCCESS, oldApplicant.get(), newApplicant.getName().fullName,
+                listingToEditApplicantFrom.getTitle()));
+
+    }
+
+    private Optional<Applicant> getOldApplicantObject(Listing listing) throws CommandException {
+        ArrayList<Applicant> applicants = listing.getApplicants();
+        Optional<Applicant> oldApplicant;
+        if (oldApplicantId.length() > 5
+                && oldApplicantId.charAt(oldApplicantId.length() - 5) == '#') {
+            String targetName = oldApplicantId.substring(0, oldApplicantId.length() - 5);
+            int targetHashCode = parseInt(oldApplicantId.substring(oldApplicantId.length() - 4));
+            oldApplicant = applicants.stream().filter(applicant -> applicant.getName().fullName.equals(targetName)
+                    && applicant.hashCode() == targetHashCode).findFirst();
+        } else {
+            List<Applicant> applicantsWithSameName = applicants.stream().filter(
+                    applicant -> applicant.getName().fullName.equals(oldApplicantId)).collect(Collectors.toList());
+            if (applicantsWithSameName.size() > 1) {
+                throw new CommandException(String.format(MESSAGE_AMBIGUOUS_APPLICANT, oldApplicantId,
+                        listing.getTitle()));
+            }
+
+            oldApplicant = applicantsWithSameName.stream().findFirst();
+        }
+        return oldApplicant;
+    }
+
+    private Listing createEditedListing(Listing listing, Applicant oldApplicant, Applicant newApplicant) {
+        ArrayList<Applicant> finalApplicants = new ArrayList<>();
+        listing.getApplicants().forEach(applicant -> {
+            if (applicant != oldApplicant) {
+                finalApplicants.add(applicant);
+            }
+        });
+        finalApplicants.add(newApplicant);
```

Comment:
I think copy the newApplicant's Name over to oldApplicant's Name instead of replacing the whole object such that the id remains the same